api/indexer/server_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package indexer
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain/chaintest"
+	"github.com/ava-labs/hypersdk/codec"
+)
+
+func TestEncodingValidate(t *testing.T) {
+	tests := []struct {
+		input  Encoding
+		exp    Encoding
+		expErr error
+	}{
+		{
+			input: JSON,
+			exp:   JSON,
+		},
+		{
+			input: Hex,
+			exp:   Hex,
+		},
+		{
+			input: Encoding(""),
+			exp:   JSON,
+		},
+		{
+			input:  Encoding("another"),
+			expErr: ErrInvalidEncodingParameter,
+		},
+	}
+
+	for _, test := range tests {
+		err := test.input.Validate()
+		if test.expErr != nil {
+			require.EqualError(t, ErrInvalidEncodingParameter, err.Error())
+		} else {
+			require.Equal(t, test.exp, test.input)
+		}
+	}
+}
+
+func TestBlockResponse(t *testing.T) {
```

Comment:
There are two ways we could go testing here imo:
- test the server/client w/ mocked indexer using httptest and test the indexer separately
- test the indexer + server/client solely through the server/client

Could we add basic tests here that use `httptest` and the server/client with a mocked of the indexer (will need to change indexer to interface as it's used in server/client)?

Reply:
We should also avoid this pattern of using `t.Run` within a single unit test. I think this makes sense for table tests (and this could be changed to a table test imo), but less so in this pattern with mostly independent test logic:

```golang
t.Run("JSON", func(t *testing.T) {
	// independent test logic
})
t.Run("Hex", func(t *testing.T) {
	// independent test logic
})
```

Reply:
Good idea by the way! The client was broken because of my changes !